src/Nethermind/Nethermind.Core/Collections/ArrayPoolList.cs

```diff
@@ -148,6 +148,7 @@ private void GuardResize(int itemsToAdd = 1)
         if (newCount > _capacity)
         {
             int newCapacity = _capacity * 2;
+            if (newCapacity == 0) newCapacity = 1;
```

Comment:
hm... newCapacity can be 0 only when original _capacity was 0? This looks weird. Maybe we should have some Min on original capacity to be at least 4 or 8?

Reply:
Could be, but then when remove maybe the newCapacity can still be zero.

Reply:
Count can be 0, capacity can't